src/opnsense/scripts/dhcp/get_leases6.py

```diff
@@ -0,0 +1,186 @@
+    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+
+    --------------------------------------------------------------------------------------
+    list dhcpv6 leases
+"""
+import sys
+sys.path.insert(0, "/usr/local/opnsense/site-python")
+import ujson
+import calendar
+import datetime
+import params
+import time
+
+def parse_date(ymd, hms):
+    result = None
+    dt = '%s %s' % (ymd, hms)
+    try:
+        result = calendar.timegm(datetime.datetime.strptime(dt, "%Y/%m/%d %H:%M:%S;").timetuple())
+    except ValueError:
+        result = None
+
+    return result
+
+def parse_iaaddr_iaprefix(input):
+    """
+    parse either an iaaddr or iaprefix segment. return a tuple
+    containing the type parsed and the corresponding segment
+    """
+    seg_type = input[0].split()[0]
+    segment = dict()
+    segment[seg_type] = input[0].split()[1]
+    for line in input[1:]:
+        parts = line.split()
+        field_name = parts[0]
+        field_value = None
+        if field_name == 'binding':
+            segment[field_name] = parts[2].strip(';')
+        elif field_name in ('preferred-life', 'max-life'):
+            field_value = parts[1].strip(';')
+        elif field_name == 'ends':
+            field_value = parse_date(parts[2], parts[3])
+
+        if field_value is not None:
+            segment[field_name] = field_value
+
+    return (seg_type, segment)
+
+def parse_iaid_duid(input):
+    """
+    parse the combined IAID_DUID value. This is provided in the form
+    of ascii characters. Non-printable characters are provided as octal escapes.
+    We return the hex representation of the raw IAID_DUID value, the IAID integer,
+    as well as the separated DUID value in a three-tuple. The IAID_DUID value is
+    used to uniquely identify a lease, so this value should be used to determine the last
+    relevant entry in the leases file.
+    """
+    input = input[1:-1] # strip double quotes
+    parsed = []
+    i = 0
+    while i < len(input):
+        c = input[i]
+        if c == '\\':
+            next_c = input[i + 1]
+            if next_c == '\\' or next == '"':
+                parsed.append("%02x" % ord(next_c))
+                i += 1
+            elif next_c.isnumeric():
+                octal_to_decimal = int(input[i+1:i+4], 8)
+                parsed.append("%02x" % octal_to_decimal)
+                i += 3
+        else:
+            parsed.append("%02x" % ord(c))
+        i += 1
+
+    iaid = int(''.join(reversed(parsed[0:4])))
+    duid = ":".join([str(a) for a in parsed[4:]])
+    iaid_duid = ":".join([str(a) for a in parsed])
+    return (iaid_duid, iaid, duid)
```

Comment:
I think I would return a `dict` here, so you can update the lease directly (`lease.update()`)